test/snapshots/linux/snapTest.js.snap

```diff
@@ -342,7 +340,144 @@ Object {
   "confinement": "strict",
   "description": "Test Application (test quite “ #378)",
   "grade": "stable",
+  "icon": "snap/gui/icon.png",
   "name": "sep",
+  "parts": Object {
```

Comment:
This is why `delete snap.parts` needs to be higher up in the logic flow. This addition shouldn't be here.

Reply:
The reason I saw `snap.parts` deletion doesn't need to move up is because if you look at everywhere the `snap` variable is used it is only these lines:

```typescript
snap.icon = "snap/gui/icon.png"
...
const desktopFile = path.join(snapMetaDir, "gui", `${snap.name}.desktop`)
...
if (this.isElectronVersionGreaterOrEqualThan("5.0.0") && !isBrowserSandboxAllowed(snap)) {
...
if (snap.compression != null) {
  args.push("--compression", snap.compression)
}
...
if (packager.packagerOptions.effectiveOptionComputed != null && (await packager.packagerOptions.effectiveOptionComputed({ snap, desktopFile, args }))) {
  return
}
```

The code that strips out the snap fields is now above that bottom one (that is what was causing all the tests to fail previously), and none of the others look at `snap.parts` at all (`isBrowserSandboxAllowed` only looks at `snap.plugs`).

Nevertheless I just added that `snap.parts` removal code back in and re-ran `TEST_FILES=snapTest UPDATE_SNAPSHOT=true pnpm test` and the snapshots do not change.

Reply:
Ah right, we explicitly set `useTemplateApp: false` to account for the no template flow w/ compression